src/lib/operators/join_index.cpp

```diff
@@ -330,8 +333,8 @@ void JoinIndex::_reference_join_two_segments_using_index(
 }
 
 template <typename SegmentPosition>
-const std::vector<IndexRange> JoinIndex::_index_ranges_for_value(SegmentPosition probe_side_position,
-                                                                 const std::shared_ptr<AbstractIndex>& index) const {
+std::vector<IndexRange> JoinIndex::_index_ranges_for_value(SegmentPosition probe_side_position,
```

Comment:
Can't `probe_side_position` not be const as well?

Reply:
Absolutely. It's just that clang-tidy did not complain about it